src/test/java/com/haxwell/apps/quizki/services/TopicServiceImplTest.java

```diff
@@ -98,6 +112,56 @@ public void test_createNew_whenAListOfTopicsThatDoesNotAlreadyExistIsPassedIn()
 		assertTrue(rtn.equals("[" + ID_TOPIC_1 + "," + ID_TOPIC_2 + "]"));
 	}
 
+	@Test
+	public void test_getTopic_whenCorrectParamsArePassedIn() {
+
+		String q = "biology";
+		int page = 1;
+		int size = 10;
+
+		//Page<Topic> pageTopic = any(Page.class);
+		Pageable pageRequest = PageRequest.of(page,size);
+
+		List<Topic> list = new ArrayList<>();
+
+		list.add(new Topic(TEXT_TOPIC_1));
+		list.add(new Topic(TEXT_TOPIC_2));
+
+		Topic fullTopic1 = new Topic(TEXT_TOPIC_1);
+		fullTopic1.setId(ID_TOPIC_1);
+
+
+		when(
+			topicRepository.findByText(q, pageRequest))
+		.thenReturn(list);
+
+		List<Topic> listResponse = topicService.getTopicByText(q, page, size);
+
+		verify(topicRepository, times(1)).findByText(q, PageRequest.of(page-1,size));
+		verify(topicRepository, times(0)).findAll(PageRequest.of(page-1,size));
+		//assertThat(listResponse.size(), equalTo(list.size()));
+	}
+
+	@Ignore("Null pointer exception with String q")
+	@Test
+	public void test_getTopic_whenNoQParamArePassedIn() {
+		String q = "";
+		int page = 1;
+		int size = 10;
+		Page<Topic> pageTopic = mock(Page.class);
+		Pageable pageRequest = PageRequest.of(page,size);
+
+		when(
+			topicRepository.findAll(pageRequest))
+		.thenReturn(pageTopic);
+
+		List<Topic> listResponse = topicService.getTopicByText("", page, size);
```

Comment:
This doesn't seem right, the `when()` returns a `Page` but `listResponse` is a `List<Topic>` type.